arch/arm/core/cortex_m/pm_s2ram.S

```diff
@@ -64,8 +64,11 @@ SECTION_FUNC(TEXT, arch_pm_s2ram_suspend)
 
 	/*
 	 * Mark entering suspend to RAM.
+	 * Store system_off on stack as r0 might be used in pm_s2ram_mark_set.
 	 */
+	push	{r0}
```

Comment:
The stack needs to remain aligned to 8bytes. Please push and pop an extra register (doesn't really matter which).

Reply:
> Out of curiosity: why does it need to be aligned to 8 bytes / 64 bit?

If I am not mistaken, a call to a function is a public interface and therefore must comply with [rule 6.2.1.2](https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst#6212stack-constraints-at-a-public-interface)

> Should these lines also be fixed?

Most likely. The current s2ram suffers from several caveat.

Reply:
Interesting. I was not aware of this. 

Just pushed an update in which I also fixed the other push/pop instructions.